### PR TITLE
ENH: Add CircleCI

### PR DESCRIPTION
**Pull request by Eric89GXL**
This produces artifacts like:

https://13-39034777-gh.circle-artifacts.com/0/home/ubuntu/sphinx-gallery/doc/_build/html/index.html

Just 1) needs to be enabled by an admin on CircleCI, and 2) turn on permissive builds for PRs (probably).

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    - sed -i "s/ENABLE_USER_SITE = .*/ENABLE_USER_SITE = False/g" /home/ubuntu/miniconda/envs/circleenv/lib/python2.7/site.py
+    - conda install -n circleenv --yes numpy scipy seaborn mayavi setuptools matplotlib pillow sphinx nose
+    - pip install -r requirements.txt
+
+  override:
+    - python setup.py develop
+
+test:
+  override:
+    - cd doc; make html
```
Comment (Titan-C) [inline circle.yml]: use
make html theme=rtd

That way we build the read the docs theme, and can preview what goes online.

File: circle.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    # Disable pyenv (no cleaner way provided by CircleCI as it prepends pyenv version to PATH)
+    - rm -rf ~/.pyenv
+    - rm -rf ~/virtualenvs
+    - wget -q http://repo.continuum.io/miniconda/Miniconda-latest-Linux-x86_64.sh -O ~/miniconda.sh
+    - chmod +x ~/miniconda.sh
+    - ~/miniconda.sh -b -p /home/ubuntu/miniconda
+    - conda update --yes --quiet conda
+    - conda create -n circleenv --yes pip python=2.7
+    - sed -i "s/ENABLE_USER_SITE = .*/ENABLE_USER_SITE = False/g" /home/ubuntu/miniconda/envs/circleenv/lib/python2.7/site.py
+    - conda install -n circleenv --yes numpy scipy seaborn mayavi setuptools matplotlib pillow sphinx nose
```
Comment (Titan-C) [inline circle.yml]: Include `sphinx_rtd_theme` as a dependency so that we can see the read the docs theme

**Comment by Eric89GXL** [comment]
@Titan-C it does not appear to be building my PR. Did you turn on permissive builds using BUILD SETTINGS -> Advanced Settings -> Build forked pull requests?

**Comment by Titan-C** [comment]
@Eric89GXL Build forked pull requests should be enabled now.

**Comment by Eric89GXL** [comment]
@Titan-C all good

https://2-25860190-gh.circle-artifacts.com/0/home/ubuntu/sphinx-gallery/doc/_build/html/index.html

**Comment by Titan-C** [comment]
It looks great. I'm even surprised it builds the mayavi gallery without any extra configuration to have an X server like we do on travis. We don't include the mayavi gallery in read the docs because the don't allow for an X server.

**Comment by Eric89GXL** [comment]
I don't know why that works, actually! For MNE we spin up an xvfb but I guess we shouldn't need to (maybe they do it automatically actually).

Might be worth switching to CircleCI. We push our docs using it.

**Comment by Eric89GXL** [comment]
See:

https://github.com/mne-tools/mne-python/blob/master/circle.yml#L142

**Comment by lesteve** [comment]
Great stuff, @Eric89GXL, thanks a lot for this!

**Comment by GaelVaroquaux** [comment]
@Eric89GXL, you've been really helpful with sphinx-gallery. Thanks a lot!

**Comment by Titan-C** [comment]
Thanks a lot @Eric89GXL

**Comment by Eric89GXL** [comment]
My pleasure, it's made doc building a lot more convenient for us

**Pull request merged**